### PR TITLE
Test MOAR GPU stuff (add the cuda demo suite!)

### PR DESCRIPTION
**Pull request by dims**
helps add the `vectorAdd` we used to have before. Taking the opportunity to run a few more things as well to ensure the GPU is working well.

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/node/gpu.go [hunk 1]
```diff
@@ -178,7 +179,20 @@ func testNvidiaCLIPod() *v1.Pod {
...
-					Image: "nvidia/cuda:12.3.2-runtime-ubuntu22.04",
+					Image: "nvidia/cuda:12.5.0-devel-ubuntu22.04",
+					Command: []string{
+						"bash",
+						"-c",
+						`
+nvidia-smi
+apt-get update -y && \
+	DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated cuda-demo-suite-12-5
+/usr/local/cuda/extras/demo_suite/deviceQuery
```
Comment (hakuna-matatah) [inline test/e2e/node/gpu.go:190]: Given you have added these additional stuff, wondering if we want to check these demo stuff in logs ?

`- gomega.Expect(log).To(gomega.ContainSubstring("CUDA Capability")) ` etc  for these additional stuff ?

**Comment by hakuna-matatah** [comment]
/lgtm

Thank you.

File: test/e2e/node/gpu.go (hunk 1 above)
Comment (hakuna-matatah) [reply test/e2e/node/gpu.go:190]: (as you iterate on this in future versions)

Comment (dims) [reply test/e2e/node/gpu.go:190]: sounds good!

Comment (hakuna-matatah) [reply test/e2e/node/gpu.go:190]: https://github.com/NVIDIA/cuda-samples/blob/master/Samples/1_Utilities/deviceQuery/deviceQuery.cpp#L109

**Pull request merged**